lib/ansible/modules/network/cloudengine/ce_vxlan_arp.py

```diff
@@ -0,0 +1,695 @@
+
+        cmd = "arp collect host enable"
+        exist = is_config_exist(self.config, cmd)
+        if self.arp_collect_host == "enable" and not exist:
+            self.cli_add_command("interface %s" %
+                                 self.vbdif_name.lower().capitalize())
+            self.cli_add_command(cmd)
+            self.cli_add_command("quit")
+        elif self.arp_collect_host == "disable" and exist:
+            self.cli_add_command("interface %s" %
+                                 self.vbdif_name.lower().capitalize())
+            self.cli_add_command(cmd, undo=True)
+            self.cli_add_command("quit")
+
+    def config_host_collect_protocal(self):
+        """Enable EVN BGP or BGP EVPN to advertise host information"""
+
+        # [undo] host collect protocol bgp
+        cmd = "host collect protocol bgp"
+        exist = is_config_exist(self.config, cmd)
+
+        if self.state == "present":
+            if self.host_collect_protocol == "bgp" and not exist:
+                self.cli_add_command(cmd)
+            elif self.host_collect_protocol == "none" and exist:
+                self.cli_add_command(cmd, undo=True)
+        else:
+            if self.host_collect_protocol == "bgp" and exist:
+                self.cli_add_command(cmd, undo=True)
+
+    def is_valid_vbdif(self, ifname):
+        """check is interface vbdif is valid"""
+
+        if not ifname.upper().startswith('VBDIF'):
+            return False
+        bdid = self.vbdif_name.replace(" ", "").upper().replace("VBDIF", "")
+        if not bdid.isdigit():
+            return False
+        if int(bdid) < 1 or int(bdid) > 16777215:
+            return False
+
+        return True
+
+    def check_params(self):
+        """Check all input params"""
+
+        # bridge domain id check
+        if self.bridge_domain_id:
+            if not self.bridge_domain_id.isdigit():
+                self.module.fail_json(
+                    msg="Error: Bridge domain id is not digit.")
+            if int(self.bridge_domain_id) < 1 or int(self.bridge_domain_id) > 16777215:
+                self.module.fail_json(
+                    msg="Error: Bridge domain id is not in the range from 1 to 16777215.")
+
+        # evn_source_ip check
+        if self.evn_source_ip:
+            if not is_valid_v4addr(self.evn_source_ip):
+                self.module.fail_json(msg="Error: evn_source_ip is invalid.")
+
+        # evn_peer_ip check
+        if self.evn_peer_ip:
+            if not is_valid_v4addr(self.evn_peer_ip):
+                self.module.fail_json(msg="Error: evn_peer_ip is invalid.")
+
+        # vbdif_name check
+        if self.vbdif_name:
+            self.vbdif_name = self.vbdif_name.replace(
+                " ", "").lower().capitalize()
+            if not self.is_valid_vbdif(self.vbdif_name):
+                self.module.fail_json(msg="Error: vbdif_name is invalid.")
+
+        # vbdif_name and arp_collect_host must set at the same time
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```